zarrs/src/array.rs

```diff
@@ -784,7 +784,9 @@ impl<TStorage: ?Sized> Array<TStorage> {
     /// Returns [`ArrayError::InvalidChunkGridIndicesError`] if the `chunk_indices` are incompatible with the chunk grid.
     pub fn chunk_subset_bounded(&self, chunk_indices: &[u64]) -> Result<ArraySubset, ArrayError> {
         let chunk_subset = self.chunk_subset(chunk_indices)?;
-        Ok(unsafe { chunk_subset.bound_unchecked(self.shape()) })
+        chunk_subset
+            .bound(self.shape())
+            .map_err(std::convert::Into::into)
```

Comment:
Would be happy to learn if there was a cleaner way of going from "enum with two values into enum with many more values, including those two"